src/amr/tagging/default_hybrid_tagger_strategy.hpp

```diff
@@ -1,85 +1,111 @@
-                Bzavg   = 0.2 * (Bz(ix - 2) + Bz(ix - 1) + Bz(ix) + Bz(ix + 1) + Bz(ix + 2));
-                Navg    = 0.2 * (N(ix - 2) + N(ix - 1) + N(ix) + N(ix + 1) + N(ix + 2));
-                Byavgp1 = 0.2 * (By(ix - 1) + By(ix) + By(ix + 1) + By(ix + 2) + By(ix + 3));
-                Bzavgp1 = 0.2 * (Bz(ix - 1) + Bz(ix) + Bz(ix + 1) + Bz(ix + 2) + Bz(ix + 3));
-                Navgp1  = 0.2 * (N(ix - 1) + N(ix) + N(ix + 1) + N(ix + 2) + N(ix + 3));
-            }
-            else
-            {
-                Byavg   = 1. / 3. * (By(ix - 1) + By(ix) + By(ix + 1));
-                Bzavg   = 1. / 3. * (Bz(ix - 1) + Bz(ix) + Bz(ix + 1));
-                Navg    = 1. / 3. * (N(ix - 1) + N(ix) + N(ix + 1));
-                Byavgp1 = 1. / 3. * (By(ix) + By(ix + 1) + By(ix + 2));
-                Bzavgp1 = 1. / 3. * (Bz(ix) + Bz(ix + 1) + Bz(ix + 2));
-                Navgp1  = 1. / 3. * (N(ix) + N(ix + 1) + N(ix + 2));
-            }
-            auto criter_by = std::abs(Byavgp1 - Byavg) / (1 + std::abs(Byavg));
-            auto criter_bz = std::abs(Bzavgp1 - Bzavg) / (1 + std::abs(Bzavg));
-            auto criter_b  = std::sqrt(criter_by * criter_by + criter_bz * criter_bz);
-            auto criter_n  = std::abs(Navgp1 - Navg) / (1 + std::abs(Navg));
-
-            auto criter = std::max(criter_b, criter_n);
-
-            if (criter > threshold)
-            {
-                tags[iCell] = 1;
-            }
-            else
-                tags[iCell] = 0;
-        }
-    }
-}
-} // namespace PHARE::amr
-
-#endif // LOHNER_HYBRID_TAGGER_STRATEGY_H
+#ifndef DEFAULT_HYBRID_TAGGER_STRATEGY_H
+#define DEFAULT_HYBRID_TAGGER_STRATEGY_H
+
+#include "hybrid_tagger_strategy.hpp"
+#include "core/data/grid/gridlayoutdefs.hpp"
+#include "core/data/vecfield/vecfield_component.hpp"
+#include "core/data/ndarray/ndarray_vector.hpp"
+#include <cstddef>
+
+
+namespace PHARE::amr
+{
+template<typename HybridModel>
+class DefaultHybridTaggerStrategy : public HybridTaggerStrategy<HybridModel>
+{
+    using gridlayout_type           = typename HybridModel::gridlayout_type;
+    static auto constexpr dimension = HybridModel::dimension;
+
+public:
+    void tag(HybridModel& model, gridlayout_type const& layout, int* tags) const override;
+};
+
+template<typename HybridModel>
+void DefaultHybridTaggerStrategy<HybridModel>::tag(HybridModel& model,
+                                                   gridlayout_type const& layout, int* tags) const
+{
+    auto& Bx = model.state.electromag.B.getComponent(PHARE::core::Component::X);
+    auto& By = model.state.electromag.B.getComponent(PHARE::core::Component::Y);
+    auto& Bz = model.state.electromag.B.getComponent(PHARE::core::Component::Z);
+
+    auto& N = model.state.ions.density();
+
+    double threshold = 0.1;
+
+    // we loop on cell indexes for all qties regardless of their centering
+    auto const& [start_x, start_y]
```

Comment:
`start_y` ?

Reply:
good catch, should be `end_x`, apparently it does not matter since endCell_x is defined just after and that's the variable that's used...... I'll fix it anyway.